pkg/config/config.go

```diff
@@ -0,0 +1,288 @@
+//
+// Copyright (c) 2017 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Red Hat trademarks are not licensed under Apache License, Version 2.
+// No permission is granted to use or replicate Red Hat trademarks that
+// are incorporated in this software or its documentation.
+//
+
+package config
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+	"sync"
+
+	logging "github.com/op/go-logging"
+	yaml "gopkg.in/yaml.v1"
+)
+
+var (
+	log = logging.MustGetLogger("config")
+)
+
```

Comment:
There might be an easier way to share the config data.
```go
// pkg/config/broker.go
client.NewConfig(...)


// pkg/config/config.go
var configFile map[string]interface{}

func NewConfig(conf map[string]interface{}) error {
       ...
       configFile = conf
       return nil
}

func GetConfigValue(value string) (string, error) {
       ....
       return configFile[value], nil
}
```
Then every pkg can have access to the var through the ```GetConfigValue``` function. If there's nested config values we could also add a ```GetNestedConfigValue```.
```go
config.GetConfigValue(configValue)
```